org.eclipse.scanning.api/src/org/eclipse/scanning/api/IModelProvider.java

```diff
@@ -19,7 +19,34 @@
  *
  * @param <T>
  */
+@FunctionalInterface
 public interface IModelProvider<T> {
 
-	public T getModel();
+	/**
+	 * 
+	 * @return the model that we are providing
+	 */
+	public T getModel() throws Exception;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)